src/components/dms/MessageMenu.tsx

```diff
@@ -28,18 +32,26 @@ export let MessageMenu = ({
   const {_} = useLingui()
   const t = useTheme()
   const {currentAccount} = useSession()
+  const chat = useChat()
   const deleteControl = usePromptControl()
 
   const isFromSelf = message.sender?.did === currentAccount?.did
 
   const onDelete = React.useCallback(() => {
-    // TODO delete the message
-  }, [])
+    if (chat.status !== ConvoStatus.Ready) return
+
+    chat.deleteMessage(message.id)
```

Comment:
This can throw FYI. It optimistically removes the message immediately, but if it fails, it's re-inserted immediately too. My thinking is that we want to handle in situ and alert the user so they can retry right from the dialog.

Reply:
so delete message first and then close the dialog?

Reply:
Yeah could just wait for it to delete, show error if fails, then close

Reply:
Could consider other options for sure, but if someone wants to delete a message, we should make sure they know when it fails

Reply:
cool, sg. will think about it today.